### ISSUE TITLE
Data interval/resolution issue introduced in 8.5.10 using graphite datasource

### ISSUE THREAD

**Issue by rschitz** [grafana/grafana#78973]
### What happened?

I know 8.5.27 is not technically supported anymore but we need time to migrate so we'd like to use it but we found an issue and were able to track down to version 8.5.10 meaning 8.5.9 was acting fine.

Here is the problem using the same dataset, in 8.5.9 the resolution match the storage schema but not 8.5.10 and above:

![287287247-01b916c5-9dcf-4326-acb3-07892b4795b3](https://github.com/grafana/grafana/assets/8385646/3557d297-a091-46eb-b5f7-14d492b38de3)

![287296383-2fffb992-401c-452d-98c1-516662edb441](https://github.com/grafana/grafana/assets/8385646/15fab70f-e682-4b9a-930d-b0a80bc3961f)

![287102708-f38f8b40-7c2f-46c0-92ef-bee4faa69595](https://github.com/grafana/grafana/assets/8385646/ff499796-37b9-4172-a73d-38a2b39d8e67)

![287103047-8697563e-9d15-4fe6-840d-da96a5541569](https://github.com/grafana/grafana/assets/8385646/5f79be17-8a27-42aa-bcd6-8261c676abef)

I'm not asking for a fix bug if this is code line i could change or a file i could replace to use the 8.5.27 and still be compliant with security it would be amazing.

ref : https://github.com/sexibytes/sexigraf/issues/366

Thanks again for this awesome tool, we are using it in our monitoring appliance project SexiGraf for almost 10 years now :)

### What did you expect to happen?

data resolution should match graphite storage schema

### Did this work before?

before 8.5.10

### How do we reproduce it?

upgrade to 8.5.10 or above

### Grafana platform?

None

### Datasource(s)?

Graphite 1.2

**Comment by bohandley** [comment]
Hi @rschitz! Thank you for researching this and thank you for your support for Grafana!

You are correct as we are not supporting versions this far back, but I was able to isolate a single change in Graphite between v8.5.9 and v8.5.10. I was not able to reproduce the  issue but I hope this helps!

https://github.com/grafana/grafana/pull/51623

**Comment by rschitz** [comment]
Hi @bohandley thanks a lot for your time!
i think i already tried to revert those changes in the files but i'll try again and let you know.

**Comment by rschitz** [comment]
Tried those files https://github.com/grafana/grafana/pull/51623/files but nothing append. Am I supposed to do something after the modifications other than a service restart? (tried a reboot)
Thanks for your support, i'll soon check if this also affect 9.x and 10.x

**Comment by rschitz** [comment]
@bohandley would you like to have access to a 8.5.9 and a 8.5.27 with the same dataset?

**Comment by bohandley** [comment]
Sure! I can't promise anything though. I looked at the changes from v8.5.9 to v8.5.27 and I wasn't able to find any meaningful changes in Graphite data source other than what I showed you. There have been no meaningful changes in the backend which is where I would expect the change would come from. For these reasons, I am not sure this is in the Graphite data source in grafana. 

If you could share that access as well as any info on your Graphite configuration in Grafana. Nothing changed with that, correct? Are you using a specific version? MetricTank or default?

**Comment by rschitz** [comment]
I'll send you access to both env on you email soon, thanks a lot for your time!
I'm using default install, i can also share the packet files used to deploy the vms.

**Comment by rschitz** [comment]
@bohandley did you get my email?

**Comment by rschitz** [comment]
Great, thank you. Just wanted to be sure it didnt end up in a spam folder or something.
Again, Thanks a lot for your help.

**Comment by alyssabull** [comment]
Hi @rschitz - have you tried adjusting the `Max data points` in Query options? This might solve your issue. 

![Image](https://github.com/user-attachments/assets/402888fa-822d-47c1-8937-9b8520b958d4)

**Comment by rschitz** [comment]
> Hi @rschitz - have you tried adjusting the `Max data points` in Query options? This might solve your issue.
> 
Hi @alyssabull and thank you for your advise, on my dashboards it's on auto, i don't know if tha'ts relevant or not.

Since i discovered this bug i managed to migrate to 9.5 so unless you guys have an interest into figuring out this bug i'd close the issue.

**Comment by rschitz** [comment]
Hi @alyssabull would you please re-open this as i found the same issue in v9.5.21:

![Image](https://github.com/user-attachments/assets/56d97ec6-a235-4e2c-a4e5-7c8b8803b375)

![Image](https://github.com/user-attachments/assets/e240e23f-4e90-46ed-bda2-6849c1099713)

![Image](https://github.com/user-attachments/assets/83d78c77-9f82-4d76-8965-1ec006b7d02d)

retentions = 5m:24h,10m:48h,30m:96h,60m:7d,240m:30d,720m:90d,2880m:1y,5760m:2y,17280m:5y

**Comment by rschitz** [comment]
tested with v11.2.0 same issue, also tested with UTC forced, same issue

**Comment by rschitz** [comment]
I just found that by changing someting in the timepicker it changes the range:
https://github.com/user-attachments/assets/cabe0e68-14ee-4ac5-8930-a824edba2ad4

**Comment by alyssabull** [comment]
Hi @rschitz - I will look into this. Thank you!

**Comment by rschitz** [comment]
> Hi [@rschitz](https://github.com/rschitz) - I will look into this. Thank you!

thank you very much @alyssabull

**Comment by rschitz** [comment]
8.5.9 vs 9.5.21 same dataset

![Image](https://github.com/user-attachments/assets/08781033-80ca-4932-984b-4dc9c0e566ca)

![Image](https://github.com/user-attachments/assets/2567f886-a660-472f-abac-53812719fcba)

@alyssabull could you re-open the issue please?

**Comment by rschitz** [comment]
@aangelisc @bohandley @tonypowa hi guys, can someone at least re-open the ticket since that's still a thing in v11 please?

**Comment by rschitz** [comment]
> Hi [@rschitz](https://github.com/rschitz), I've reopened the issue. Are you able to provide us with a dashboard that we can use to reproduce?

Hi @aangelisc i didn't get the notification for your message, my deepest apologies.
I can provide you a dashboard and some wisper files, please send me an email to raphael[at]schitz[dot]net

**Comment by aangelisc** [comment]
Hi @rschitz, apologies again for the delayed response. Are you able to share the dashboard here please? I'm a little uncertain what the issue actually is so steps to reproduce would be very helpful!

**Comment by rschitz** [comment]
> Hi [@rschitz](https://github.com/rschitz), apologies again for the delayed response. Are you able to share the dashboard here please? I'm a little uncertain what the issue actually is so steps to reproduce would be very helpful!

I can share you access (http and ssh) of 2 vms with 2 different install so you can check whatever you need, that's why i asked you to send me an email

**Comment by aangelisc** [comment]
Hi @rschitz, what is your email?

I'm curious if #107414 may solve this problem - previously the date object was being mutated but we've recently fixed that. Are you able to test with the nightly docker build?

**Comment by rschitz** [comment]
> Hi [@rschitz](https://github.com/rschitz), what is your email?
> 
> I'm curious if [#107414](https://github.com/grafana/grafana/pull/107414) may solve this problem - previously the date object was being mutated but we've recently fixed that. Are you able to test with the nightly docker build?

raphael[at]schitz[dot]net

i'll try with the nightly docker build, thanks a lot for your help

**Comment by aangelisc** [comment]
Hey @rschitz, did the nightly build solve your problem?

**Comment by rschitz** [comment]
> Hey [@rschitz](https://github.com/rschitz), did the nightly build solve your problem?

sorry i didnt have time to test it yet. will try my best in the next days.

**Comment by rschitz** [comment]
> Hey [@rschitz](https://github.com/rschitz), did the nightly build solve your problem?

sorry for the delay, same issue

<img width="1586" height="692" alt="Image" src="https://github.com/user-attachments/assets/29b83685-8f57-4af6-9b83-0e18affcf542" />

<img width="1569" height="697" alt="Image" src="https://github.com/user-attachments/assets/aa36fa25-143b-4dde-b369-a89d9c658f71" />

**Comment by aangelisc** [comment]
Hi @rschitz, apologies for the delay. 

Since I last looked at this we have introduced a backend mode for Graphite, could you please try that by enabling `graphiteBackendMode` in your Grafana config to see if that fixes the issue.

If it doesn't, could you please share a dashboard with us and outline what the expected behaviour is vs what you're currently seeing. Thanks!

**Comment by rschitz** [comment]
Thanks @aangelisc i'll try this asap, can you briefly explain what it does please?

**Comment by aangelisc** [comment]
Hi @rschitz, we've recently converted Graphite to be a backend plugin. This now routes queries via the Grafana server rather than directly from the client browser and now has some updated logic 😊

**Comment by rschitz** [comment]
@aangelisc thanks for the details, that sounds great! will it be backported to v11 by any chances?

**Comment by aangelisc** [comment]
Hi @rschitz, unfortunately not, it's quite a large refactor of the plugin.

**Comment by rschitz** [comment]
@aangelisc I understand, thanks for sharing. Is it in 12.3.3 already?

**Comment by aangelisc** [comment]
Yes it is, please give it a try and let us know if that resolves your issue 😊

**Comment by rschitz** [comment]
Same issue @aangelisc :

<img width="2694" height="1311" alt="Image" src="https://github.com/user-attachments/assets/9ce8979d-19ec-4541-823d-d82ce444e63a" />

<img width="2692" height="1309" alt="Image" src="https://github.com/user-attachments/assets/45e4be80-7182-4398-9b1d-7532ff240665" />
